src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerActor.scala

```diff
@@ -132,8 +132,8 @@ private[impl] class GroupManagerActor(
         plan = DeploymentPlan(from, to, resolve, version, toKill)
         _ = validateOrThrow(plan)(DeploymentPlan.deploymentPlanValidator(config))
         _ = log.info(s"Computed new deployment plan:\n$plan")
-        _ <- scheduler.deploy(plan, force)
         _ <- groupRepo.storeRoot(plan.target, plan.createdOrUpdatedApps, plan.deletedApps)
+        _ <- scheduler.deploy(plan, force)
```

Comment:
So actually, there is a more tricky case that we should solve then - store the rootVersion then ask the scheduler to deploy, then storeRoot (setting it to the current version). I ran into some scenarios - it could have been related to the fact that pod deserialization was busted where deployment plans were referring to roots that hadn't been stored. 

Perhaps I haven't had enough coffee yet, but I think this is still the case unless we do "storeRootVersion" before we do a scheduler deploy.